### PR TITLE
Retry queue: Connect transaction container to domain forwarder

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR is part of **persisting failed transactions on disk when in-mem buffer is full**.
See #6867 for more context.

With this PR, when the retry queue from the domain_forwarder is full, the transactions are now stored on disk.

### Additional Notes

This PR uses the code from https://github.com/DataDog/datadog-agent/pull/6886

**Reviewing commit by commit is strongly recommended.**

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: releasenotes/notes/add_forwarder_storage_max_size_in_bytes-d860cf200534df6b.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Each section from every releasenote are combined when the
```
Comment (apigirl) [inline releasenotes/notes/add_forwarder_storage_max_size_in_bytes-d860cf200534df6b.yaml:1]:
```suggestion
# Each section from every release note are combined when the
```

**Review by apigirl — COMMENTED** [review]
just one small docs feedback, otherwise lgtm

File: pkg/forwarder/domain_forwarder.go [hunk 2]
```diff
@@ -135,12 +147,39 @@ func (f *domainForwarder) retryTransactions(retryBefore time.Time) {
...
+	dropCount, err := f.optionalTransactionContainer.Add(t)
+	if err != nil {
+		log.Errorf("Error when adding a transaction to the retry queue: %v", err)
+	}
+
+	if dropCount > 0 {
+		transactionEndpointName := t.GetEndpointName()
+		transactionsDroppedByEndpoint.Add(transactionEndpointName, int64(dropCount))
+		transactionsDropped.Add(int64(dropCount))
+		tlmTxDropped.Inc(f.domain, transactionEndpointName)
```
Comment (remeh) [inline pkg/forwarder/domain_forwarder.go:164]: Shouldn't it be increased by dropCount instead of 1?

Comment (ogaca-dd) [reply pkg/forwarder/domain_forwarder.go:164]: Code updated.

File: pkg/forwarder/domain_forwarder.go [hunk 3]
```diff
@@ -36,36 +35,38 @@ type domainForwarder struct {
...
 	connectionResetInterval      time.Duration
 	internalState                uint32
 	m                            sync.Mutex // To control Start/Stop races
-
-	blockedList *blockedEndpoints
+	transactionPrioritySorter    transactionPrioritySorter
+	blockedList                  *blockedEndpoints
 }
 
-func newDomainForwarder(domain string, numberOfWorkers int, retryQueueLimit int, retryQueueAllPayloadsMaxSize int, connectionResetInterval time.Duration) *domainForwarder {
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply pkg/forwarder/domain_forwarder.go:46]: `newDomainForwarder` cannot provide a default implementation for `transactionContainer`. `nil` is a valid value for `transactionContainer`.

File: pkg/forwarder/forwarder.go [hunk 4]
```diff
@@ -229,15 +254,67 @@ func NewDefaultForwarder(options *Options) *DefaultForwarder {
...
+		outdatedFileInDays := config.Datadog.GetInt("forwarder_outdated_file_in_days")
+		var err error
+
+		optionalRemovalPolicy, err = newFailedTransactionRemovalPolicy(storagePath, outdatedFileInDays)
+		if err != nil {
+			log.Errorf("Error when initializing the removal policy: %v", err)
+		}
+	}
+
+	flushToDiskMemRatio := config.Datadog.GetFloat64("forwarder_flush_to_disk_mem_ratio")
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply pkg/forwarder/forwarder.go:273]: Code updated.

File: pkg/forwarder/domain_forwarder.go [hunk 5]
```diff
@@ -36,36 +35,38 @@ type domainForwarder struct {
 	retryQueue                   []Transaction
 	retryQueueLimit              int
 	retryQueueAllPayloadsMaxSize int
+	optionalTransactionContainer *transactionContainer
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply pkg/forwarder/domain_forwarder.go:38]: As discussed in private this code will be updated.

**Comment by ogaca-dd** [comment]
Closed in favor of https://github.com/DataDog/datadog-agent/pull/6921.

**Pull request closed without merging**